packages/m2m-gateway/src/services/purposeService.ts

```diff
@@ -182,5 +203,41 @@ export function purposeServiceBuilder(clients: PagoPAInteropBeClients) {
 
       return toM2mGatewayApiPurposeVersion(createdVersion);
     },
+    async activatePurpose(
+      purposeId: PurposeId,
+      { logger, headers }: WithLogger<M2MGatewayAppContext>
+    ): Promise<void> {
+      logger.info(
+        `Retrieveing current version for purpose ${purposeId} activation`
+      );
+      const purposeResponse = await clients.purposeProcessClient.getPurpose({
+        params: {
+          id: purposeId,
+        },
+        headers,
+      });
+
+      const versionToActivate = retrievePurposeCurrentVersion(
+        purposeResponse.data
+      );
+
+      logger.info(
+        `Activating version ${versionToActivate.id} of purpose ${purposeId}`
+      );
+
+      const { metadata } =
+        await clients.purposeProcessClient.activatePurposeVersion(undefined, {
+          params: { purposeId, versionId: versionToActivate.id },
+          headers,
+        });
+
+      await pollPurpose(
+        {
+          data: purposeResponse.data,
+          metadata,
+        },
+        headers
+      );
+    },
```

Comment:
This implementation appears incorrect. The specification states that only purposes in DRAFT state should be activated, but the current logic allows other states to pass. The function should also be named `activateDraftPurpose`